src/main/java/com/auth0/json/mgmt/guardian/TwilioFactorProvider.java

```diff
@@ -84,11 +95,15 @@ public String getMessagingServiceSID() {
      * Setter for the Twilio Messaging Service SID.
      *
      * @param messagingServiceSID the messaging service SID.
+     * @throws IllegalArgumentException when both `from` and `messagingServiceSID` are set
      * @deprecated use the constructor instead
      */
     @Deprecated
     @JsonProperty("messaging_service_sid")
-    public void setMessagingServiceSID(String messagingServiceSID) {
+    public void setMessagingServiceSID(String messagingServiceSID) throws IllegalArgumentException {
+        if (from != null) {
```

Comment:
If the user is removing the `messagingServiceSID` i.e. setting `messagingServiceSID=null` then the exception will be thrown anyway. The check should be: 

```java
if (from!=null && messagingServiceSID!=null){
    //.. 
}
```